test/integration/scheduler/scheduler_test.go

```diff
@@ -1296,6 +1313,34 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			util.ExpectWorkloadsToBePending(ctx, k8sClient, wl2)
 			util.ExpectPendingWorkloadsMetric(strictFIFOClusterQ, 0, 1)
 		})
+
+		ginkgo.It("Pending workload with StrictFIFO doesn't block other CQ from borrowing from a third CQ", func() {
```

Comment:
Let's also add a test case to cover this: https://github.com/kubernetes-sigs/kueue/issues/1283#issuecomment-1841071485

Reply:
Extended test case to cover both scenarios